cleanlab/token_classification/filter.py

```diff
@@ -1,22 +1,24 @@
 import numpy as np
 from cleanlab.filter import find_label_issues as find_label_issues_main
-from typing import Optional, Union, Tuple
+from typing import List, Tuple
 
 
 def find_label_issues(
-    labels: list, pred_probs: list, return_indices_ranked_by: str = "self_confidence"
-) -> list:
+    labels: List[List[int]],
+    pred_probs: List[np.ndarray],
```

Comment:
Are these types overly restrictive? Could the user, for example, pass in a 2d np.array of ints for `labels`, or a 3d np.array of floats for `pred_probs`?

Reply:
Slightly misaligned with [`find_label_issues`](https://docs.cleanlab.ai/v2.0.0/cleanlab/filter.html#cleanlab.filter.find_label_issues) types, where we ask for a np.array for labels.

Reply:
> Are these types overly restrictive? Could the user, for example, pass in a 2d np.array of ints for `labels`, or a 3d np.array of floats for `pred_probs`?

The only use case I can think of is if the user wants to truncate their inputs or add padding tokens to fit a batch of sentences/documents into a rectangular array.

I'm not sure if it was originally intended to support truncated/padded inputs. Thoughts on this, @ericwang1997?

> Slightly misaligned with [`find_label_issues`](https://docs.cleanlab.ai/v2.0.0/cleanlab/filter.html#cleanlab.filter.find_label_issues) types, where we ask for a np.array for labels.

Looks like both of these should use `LabelLike`.
It should be `List[LabelLike]` for token classification, but there are no tests that cover this.

Reply:
I think the two supported types should be: `List[List[int]]`, `List[np.ndarray of ints]`
@elisno Can you verify if the second type works and if not, just convert each of the arrays inside the list to a List?  

@anishathalye This does not currently work for 2d array of labels because of the variable-length nature of sentences. And dealing with padding would be annoying IMO and too much unnecessary extra work. The user anyway likely has to process their raw predictions/labels from neural net format into cleanlab's format, so we might as well ask for the simplest and most intuitive format.

Reply:
Both types work, but maybe it was a bit excessive of me to tracking the types of the nested sequences at this stage. Reverting these changes for now.